Render the angular component as the root element of the react component

jrsquared opened the pull request:
In our conversion story, all of our angular components are already wrapped in a single component and adding the wrapper div breaks a lot of our existing layout. Fits our model better to be able to just replace.

Also added tests around this new flag.

bcherny commented:
@jrsquared Thanks for the contribution! A few notes:

1. Let's make this the default behavior - no need for `replace` flag
2. Can you add a test to make sure that the react `div` is properly unmounted from the DOM when the component is destroyed?

jrsquared commented:
1. Done. Regarding #2, it actually wasn't calling $destroy because we were losing the react context on replace. So I cleaned it up and made the angular component actually be the root element of the react component.
2. Added the test.

bcherny commented on test.tsx:
```diff
@@ -31,11 +30,13 @@ const FooBarBaz = {
-    <span>{{ $ctrl.fooService.add42($ctrl.fooBar) }}</span>
-    <div ng-click="$ctrl.onClick()">Click Me</div>
-    <ng-transclude></ng-transclude>
+    <span id="foobarbaz">
```
With this PR, do component templates now need to be wrapped in an enclosing element?

jrsquared replied:
Actually, now that I switched to just use the angular component as the root, probably not. I'll take a look at this, good point.

bcherny commented on test.tsx:
```diff
@@ -64,65 +65,77 @@ it('should give a react component', () => {
   const Foo2 = compile($injector)
   const element = document.createElement('div')
-  render(<Foo2 foo='hello' fooBar={42} />, element)
+  render(<Foo2 foo='hello' fooBar={42} />, element) as Element
```
This assert doesn't seem to be used

jrsquared replied:
Yep, cruft, good catch.

bcherny commented on test.tsx:
```diff
@@ -64,65 +65,77 @@ it('should give a react component', () => {
   spyOn(FooController.prototype, '$onDestroy')
-  unmountComponentAtNode(element)
+  const unmounted = unmountComponentAtNode(element)
+  expect(unmounted).toBeTruthy
```
Typo: Be sure to invoke `toBeTruthy`

bcherny replied:
Since we're doing more magic now, can you also assert that `element` is now empty?

bcherny commented on test.tsx:
```diff
@@ -64,65 +65,77 @@ it('should give a react component', () => {
-    foo: string
-    fooBar: number
-  }
+it('should use the angular component as the root component', () => {
```
Nice

jrsquared commented:
Addressed all comments.

bcherny commented:
Thanks for the contribution @jrsquared! Published this change @2.0.0
